01-builder-visualization/exercises/twitter-dogs.md

```diff
@@ -0,0 +1,199 @@
+
+##### Deleting the heading "actor_preferredusername" and adding an @ before the user name
+
+* Find this piece of code:
+```
+<div class="CDB-infowindow-listItem CDB-infowindow-listItem--order2">
+    <h5 class="CDB-infowindow-subtitle">actor_preferredusername</h5>
+    <h4 class="CDB-infowindow-title">{{actor_preferredusername}}</h4>
+```
+and alter it as follows:
+```
+<div class="CDB-infowindow-listItem CDB-infowindow-listItem--order2">
+    <h5 class="CDB-infowindow-subtitle"></h5>
+    <h4 class="CDB-infowindow-title">@{{actor_preferredusername}}</h4>
+```
+
+##### Deleting the heading "body"
+
+* Find this piece of code:
+```
+<div class="CDB-infowindow-listItem CDB-infowindow-listItem--order3">
+<h5 class="CDB-infowindow-subtitle">body</h5>
+<h4 class="CDB-infowindow-title">{{body}}</h4>
+```
+and alter it as follows:
+```
+<div class="CDB-infowindow-listItem CDB-infowindow-listItem--order3">
+<h5 class="CDB-infowindow-subtitle"></h5>
+<h4 class="CDB-infowindow-title">{{body}}</h4>
+```
+
+##### Making a clickable hyperlink that will open the linked tweet in a new browser tab
+
+* Find this piece of code:
+```
+<div class="CDB-infowindow-listItem CDB-infowindow-listItem--order4">
+    <h5 class="CDB-infowindow-subtitle">link</h5>
+    <h4 class="CDB-infowindow-title">{{link}}</h4>
+```
+and alter it as follows:
+```
+<div class="CDB-infowindow-listItem CDB-infowindow-listItem--order4">
+    <h5 class="CDB-infowindow-subtitle"></h5>
+    <h4 class="CDB-infowindow-title"><a href="{{link}}" target="_blank">View Tweet</a></h4>
+```
+
+* Click `APPLY`.
+
+![pop-up click HTML](imgs/01-dog-05.png)
+
+<hr>
+
+## 3. Widgets <a name="widgets"></a>
+
+### 3. 1. Add widgets
+
+* Navigate to  the **WIDGETS** pane.
+  * You will notice that your time slider is already present.
+* Click `ADD`.
+  * In the options of the **CATEGORY** tab, check the *category_terms* widget. Make sure you are adding a widget from the `A0` layer node.
+  * In the options of the **FORMULA** tab, check `retweetcount` and `actor_followerscount`, again making sure you are adding the widget with the source as `A0`.
+* Click on `CONTINUE`.
+
+### 3. 2. Order and edit widgets
+
+* Navigate back to the **WIDGETS** pane.
+* **Click the *time widget*.**
+  - Change the title to **When are people tweeting?**.
+  - You can change the color of the widget bars now too if you'd like.
+
+* **Click the *retweetcount* widget.**
+  - Change the title to **How popular are these tweets?**
+  - Make sure the type is set to `FORMULA`.
+  - Change the **OPERATION** to `SUM(retweetcount)`
+  - Add the **SUFFIX**: ` retweets` to let your viewers know what the number they are looking at means.
+  - Widgets are only as good as the data you provide to them. You don't always want the viewers to interpret what each item means. Go ahead and add a description for the widget. In this case, I am going to explain the purpose of a retweet. Here's what I am saying:
```

Comment:
Add some more specifics here in case they don't understand how, something like - From the widgets pane, you can reorder the widgets and how they are displayed on the map by dragging and dropping each widget option in the panel to reorder.